r/src/scalar.cpp

```diff
@@ -68,6 +68,12 @@ SEXP Scalar__as_vector(const std::shared_ptr<arrow::Scalar>& scalar) {
   return Array__as_vector(array);
 }
 
+// [[arrow::export]]
+std::shared_ptr<arrow::Array> MakeArrayFromScalar(
```

Comment:
This ended up not being necessary to solve the issue, but it's useful so I left it in.